game/glshader.cc

```diff
@@ -25,22 +25,27 @@ namespace {
 }
 
 /// Dumps Shader/Program InfoLog
-void Shader::dumpInfoLog(GLuint id) {
+void Shader::dumpInfoLog(GLuint id) { 
 	int maxLength;
 
 	if (glIsShader(id)) glGetShaderiv(id, GL_INFO_LOG_LENGTH, &maxLength);
 	else glGetProgramiv(id, GL_INFO_LOG_LENGTH, &maxLength);
 
-	char infoLog[maxLength];
+	char *infoLog = new char[maxLength];
```

Comment:
Performous AFAIK never uses the `new` operation and certainly never the `new[]` which is just plain evil. Don't use manual memory management. Instead, in this case, use

```c++
std::vector<char> infoLog(maxLength);
// and where you need the data pointer
infoLog.data()
```

Reply:
Will change it, also based on this:
https://www.khronos.org/opengl/wiki/Shader_Compilation